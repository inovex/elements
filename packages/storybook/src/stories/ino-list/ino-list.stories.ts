import { html } from 'lit-html';
import { useEffect } from '@storybook/client-api';

export default {
  title: 'Structure/ino-list ',
  component: 'ino-list',
};

export const Playground = (args) => html`
  <ino-list
    dense="${args.dense}"
    two-lines="${args.twoLines}"
    avatar="${args.avatar}"
    >
      <ino-list-item text="First text item"></ino-list-item>
      <ino-list-item text="Second text item"></ino-list-item>
      <ino-list-item text="Third text item"></ino-list-item>
  </ino-list>
`;
Playground.args = {
  dense: false,
  twoLines: false,
  avatar: false,
}

const exampleImg = html`
  <ino-img
    slot="leading"
    src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
    ratio-width="1"
    ratio-height="1"
  ></ino-img>
`;

export const TwoLines = () => html`
  <ino-list two-lines>
    <ino-list-item text="First text item" secondary-text="Secondary"></ino-list-item>
    <ino-list-item text="Second text item" secondary-text="Secondary"></ino-list-item>
    <ino-list-item text="Third text item" secondary-text="Secondary"></ino-list-item>
  </ino-list>
`;

export const Dense = () => html`
  <h4>Default</h4>
  <ino-list dense class="first-list">
    <ino-list-item text="First text item"></ino-list-item>
    <ino-list-item text="Second text item"></ino-list-item>
    <ino-list-item text="Third text item"></ino-list-item>
  </ino-list>

  <h4>Dense and Two Lines</h4>
  <ino-list dense two-lines>
    <ino-list-item text="First text item" secondary-text="Secondary"></ino-list-item>
    <ino-list-item text="Second text item" secondary-text="Secondary"></ino-list-item>
    <ino-list-item text="Third text item" secondary-text="Secondary"></ino-list-item>
  </ino-list>
`;

export const Avatar = () => html`
  <h4>Default</h4>
  <ino-list avatar>
    <ino-list-item text="First text item">
      ${exampleImg}
    </ino-list-item>
    <ino-list-item text="Second text item">
      ${exampleImg}
    </ino-list-item>
  </ino-list>

  <h4>Avatar and Dense</h4>
  <ino-list avatar dense>
    <ino-list-item text="First text item">
      ${exampleImg}
    </ino-list-item>
    <ino-list-item text="Second text item">
      ${exampleImg}
    </ino-list-item>
  </ino-list>

  <h4>Avatar and Two Lines</h4>
  <ino-list avatar two-lines>
    <ino-list-item text="First text item" secondary-text="Secondary">
      ${exampleImg}
    </ino-list-item>
    <ino-list-item text="Second text item" secondary-text="Secondary">
      ${exampleImg}
    </ino-list-item>
  </ino-list>
`;
