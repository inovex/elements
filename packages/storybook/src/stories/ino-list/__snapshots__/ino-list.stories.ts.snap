// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Structure/ino-list  Avatar smoke-test 1`] = `
<div id="root-inner">
  <ino-list dense="false"
            two-lines="false"
            avatar="true"
  >
    <ul class="mdc-deprecated-list mdc-deprecated-list--avatar-list">
      <ino-list-item text="First text item"
                     secondary-text
      >
        <li class="mdc-deprecated-list-item">
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__graphic"
                role="presentation"
          >
            <ino-img slot="leading"
                     src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                     ratio-width="1"
                     ratio-height="1"
            >
            </ino-img>
          </span>
          <span class="mdc-deprecated-list-item__text">
            First text item
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Second text item"
                     secondary-text
      >
        <li class="mdc-deprecated-list-item">
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__graphic"
                role="presentation"
          >
            <ino-img slot="leading"
                     src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                     ratio-width="1"
                     ratio-height="1"
            >
            </ino-img>
          </span>
          <span class="mdc-deprecated-list-item__text">
            Second text item
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Third text item"
                     secondary-text
      >
        <li class="mdc-deprecated-list-item">
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__graphic"
                role="presentation"
          >
            <ino-img slot="leading"
                     src="https://cdn-images-1.medium.com/max/1600/1*HP8l7LMMt7Sh5UoO1T-yLQ.png"
                     ratio-width="1"
                     ratio-height="1"
            >
            </ino-img>
          </span>
          <span class="mdc-deprecated-list-item__text">
            Third text item
          </span>
        </li>
      </ino-list-item>
    </ul>
  </ino-list>
</div>
`;

exports[`Structure/ino-list  Dense smoke-test 1`] = `
<div id="root-inner">
  <ino-list dense="true"
            two-lines="false"
            avatar="false"
            class="hydrated"
  >
    <ul class="mdc-deprecated-list mdc-deprecated-list--dense">
      <ino-list-item text="First text item"
                     secondary-text
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            First text item
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Second text item"
                     secondary-text
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            Second text item
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Third text item"
                     secondary-text
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            Third text item
          </span>
        </li>
      </ino-list-item>
    </ul>
  </ino-list>
</div>
`;

exports[`Structure/ino-list  Playground smoke-test 1`] = `
<div id="root-inner">
  <ino-list dense="false"
            two-lines="false"
            avatar="false"
            class="hydrated"
  >
    <ul class="mdc-deprecated-list">
      <ino-list-item text="First text item"
                     secondary-text
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            First text item
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Second text item"
                     secondary-text
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            Second text item
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Third text item"
                     secondary-text
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            Third text item
          </span>
        </li>
      </ino-list-item>
    </ul>
  </ino-list>
</div>
`;

exports[`Structure/ino-list  TwoLines smoke-test 1`] = `
<div id="root-inner">
  <ino-list dense="false"
            two-lines="true"
            avatar="false"
            class="hydrated"
  >
    <ul class="mdc-deprecated-list mdc-deprecated-list--two-line">
      <ino-list-item text="First text item"
                     secondary-text="Secondary Text"
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            <span class="mdc-deprecated-list-item__primary-text">
              First text item
            </span>
            <span class="mdc-deprecated-list-item__secondary-text">
              Secondary Text
            </span>
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Second text item"
                     secondary-text="Secondary Text"
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            <span class="mdc-deprecated-list-item__primary-text">
              Second text item
            </span>
            <span class="mdc-deprecated-list-item__secondary-text">
              Secondary Text
            </span>
          </span>
        </li>
      </ino-list-item>
      <ino-list-item text="Third text item"
                     secondary-text="Secondary Text"
                     class="hydrated"
      >
        <li class="mdc-deprecated-list-item mdc-ripple-upgraded"
            tabindex="-1"
        >
          <span class="mdc-deprecated-list-item__ripple">
          </span>
          <span class="mdc-deprecated-list-item__text">
            <span class="mdc-deprecated-list-item__primary-text">
              Third text item
            </span>
            <span class="mdc-deprecated-list-item__secondary-text">
              Secondary Text
            </span>
          </span>
        </li>
      </ino-list-item>
    </ul>
  </ino-list>
</div>
`;
